app/controllers/merchant/items_controller.rb

```diff
@@ -0,0 +1,13 @@
+class Merchant::ItemsController < ApplicationController
+  before_action :require_merchant
+
+  def index
+    @items = current_user.merchants[0].items
+  end
+
+  private
+
+  def require_merchant
+    render file: "/public/404" unless current_merchant?
+  end
+end
```

Comment:
I think we are using this code in multiple places so possibly we could refactor the require_merchant method.

Reply:
Good call steve, if it is in many places it would be a good use case for putting this require_merchant in the application controller